src/test/scala/io/iohk/ethereum/validators/BlockHeaderValidatorSpec.scala

```diff
@@ -13,195 +13,183 @@ import org.scalatest.{FlatSpec, Matchers}
-      val validateResult = blockHeaderValidator.validate(blockHeader, validBlockParent.header)
+      val blockHeaderWithInvalidTimestamp = validSignedBlockHeader.header.copy(unixTimestamp = timestamp)
+      val signedBlockHeaderWithInvalidTimestamp = SignedBlockHeader.sign(blockHeaderWithInvalidTimestamp, validBlockBeneficiaryKeyPair)
+      val validateResult = blockHeaderValidator.validate(signedBlockHeaderWithInvalidTimestamp, validSignedBlockHeaderParent.header)
       timestamp match {
-        case t if t <= validBlockParent.header.unixTimestamp => assert(validateResult == Left(HeaderTimestampError))
+        case t if t <= validSignedBlockHeaderParent.header.unixTimestamp => assert(validateResult == Left(HeaderTimestampError))
         case _ => assert(validateResult == Right(BlockHeaderValid))
       }
     }
   }
 
   it should "return a failure if created based on invalid difficulty" in new TestSetup {
-    (electionManagerMock.verifyIsLeader _).expects(*,*).returning(true).anyNumberOfTimes()
+    (electionManagerMock.verifyIsLeader _).expects(*, *).returning(true).anyNumberOfTimes()
     (slotCalculatorMock.getSlotStartingMillis _).expects(*).returning(CurrentTime.toMillis - 1).anyNumberOfTimes()
     (clockMock.now _).expects().returning(CurrentTime).anyNumberOfTimes()
 
     forAll(bigIntGen) { difficulty =>
-      val blockHeader = validBlockHeader.copy(validBlockHeader.header.copy(difficulty = difficulty))
-      val validateResult = blockHeaderValidator.validate(blockHeader, validBlockParent.header)
-      if(difficulty != validBlockHeader.header.difficulty) assert(validateResult == Left(HeaderDifficultyError))
+      val blockHeader = validSignedBlockHeader.copy(validSignedBlockHeader.header.copy(difficulty = difficulty))
+      val validateResult = blockHeaderValidator.validate(blockHeader, validSignedBlockHeaderParent.header)
+      if (difficulty != validSignedBlockHeader.header.difficulty) assert(validateResult == Left(HeaderDifficultyError))
       else assert(validateResult == Right(BlockHeaderValid))
     }
   }
 
   it should "return a failure if created based on invalid gas used" in new TestSetup {
-    (electionManagerMock.verifyIsLeader _).expects(*,*).returning(true).anyNumberOfTimes()
+    (electionManagerMock.verifyIsLeader _).expects(*, *).returning(true).anyNumberOfTimes()
     (slotCalculatorMock.getSlotStartingMillis _).expects(*).returning(CurrentTime.toMillis - 1).anyNumberOfTimes()
     (clockMock.now _).expects().returning(CurrentTime).anyNumberOfTimes()
 
     forAll(bigIntGen) { gasUsed =>
-      val blockHeader = validBlockHeader.copy(validBlockHeader.header.copy(gasUsed = gasUsed))
-      val validateResult = blockHeaderValidator.validate(blockHeader, validBlockParent.header)
-      if(gasUsed > validBlockHeader.header.gasLimit) assert(validateResult == Left(HeaderGasUsedError))
+      val blockHeader = validSignedBlockHeader.copy(validSignedBlockHeader.header.copy(gasUsed = gasUsed))
```

Comment:
Should we sign this block? For example if `bigIntGen` is 0 the signature validation fails.